crates/thetawave_interface/src/states.rs

```diff
@@ -20,3 +20,24 @@ pub enum GameStates {
     Playing,
     Paused,
 }
+
+#[derive(Component)]
+pub struct MainMenuCleanup;
+
+#[derive(Component)]
+pub struct GameCleanup;
+
+#[derive(Component)]
+pub struct GameOverCleanup;
+
+#[derive(Component)]
+pub struct VictoryCleanup;
+
+#[derive(Component)]
+pub struct PauseCleanup;
+
+#[derive(Component)]
+pub struct InstructionsCleanup;
+
+#[derive(Component)]
+pub struct CharacterSelectionCleanup;
```

Comment:
Maybe 

```rust
pub struct CleanupTag(AppState)
```
or [bit flags](https://stackoverflow.com/questions/43509560/how-to-implement-bitwise-operations-on-a-bitflags-enum) if you want to assign multiple cleanup tags (and back it by a u8)
or move these to a module like `cleanup_tags` with a top level doc like 

```
Minimal indicators for when attributes should be despawned from the game. 
```

Reply:
I attempted to generalize the cleanup tag, but was not successful. I think this is still the best way of doing it. I'll move them to a module like you suggested though.